lib/ansible/plugins/connection/network_cli.py

```diff
@@ -275,27 +279,24 @@ def exec_command(self, cmd):
             else:
                 # Prompt was a Sequence of strings.  Make sure they're byte strings
                 obj['prompt'] = [to_bytes(p, errors='surrogate_or_strict') for p in obj['prompt'] if p is not None]
-        if obj['command'] == b'close_shell()':
-            return self.close_shell()
-        elif obj['command'] == b'open_shell()':
-            return self.open_shell()
-        elif obj['command'] == b'prompt()':
-            return (0, self._matched_prompt, b'')
 
-        try:
-            if self._shell is None:
-                self.open_shell()
-        except AnsibleConnectionFailure as exc:
-            # FIXME: Feels like we should raise this rather than return it
-            return (1, b'', to_bytes(exc))
+        if 'jsonrpc' in obj:
+            if self._cliconf:
+                out = self._exec_rpc(obj)
+            else:
+                out = self.internal_error("cliconf is not supported for network_os %s" % self._play_context.network_os)
+            return 0, to_bytes(out, errors='surrogate_or_strict'), b''
+
+        if obj['command'] == b'prompt()':
+            return 0, self._matched_prompt, b''
 
         try:
             if not signal.getsignal(signal.SIGALRM):
                 signal.signal(signal.SIGALRM, self.alarm_handler)
             signal.alarm(self._play_context.timeout)
-            out = self.send(obj)
+            out = self.send(obj['command'], obj.get('prompt'), obj.get('answer'), obj.get('sendonly'))
             signal.alarm(0)
-            return (0, out, b'')
+            return 0, out, b''
         except (AnsibleConnectionFailure, ValueError) as exc:
             # FIXME: Feels like we should raise this rather than return it
```

Comment:
raising an exception here does you no good since this is probably running as a background process